time_tracker_api/time_entries/time_entries_model.py

```diff
@@ -74,6 +74,35 @@ def create_sql_ignore_id_condition(id: str):
         else:
             return "AND c.id!=@ignore_id"
 
+    @staticmethod
+    def create_sql_date_range_filter(custom_args: dict) -> str:
+        if 'start_date' and 'end_date' in custom_args:
+            return """
+            ((c.start_date BETWEEN @start_date AND @end_date) OR
+             (c.end_date BETWEEN @start_date AND @end_date))
+            """
+        else:
+            return ''
+
+    def find_all(self, partition_key_value: str, conditions: dict, custom_args: dict):
+        custom_conditions = []
+        custom_conditions.append(
+            self.create_sql_date_range_filter(custom_args)
+        )
+
+        custom_params = [
```

Comment:
You should have a function that knows the params logic for this particular case. So this function will create the list of dictionary entries you need to add for the query to run. That logic should not be inside of this function `find_all`.